lib/kafka_rest_client.js

```diff
@@ -133,33 +135,55 @@ KafkaRestClient.prototype.initHttpClient = function initHttpClient(options) {
 
 KafkaRestClient.prototype.initLineageOption = function initLineageOption(options) {
     var self = this;
-    if ('applicationName' in options) {
-        self.applicationName = options.applicationName; // the application name for data lineage
+    if ('serviceName' in options) {
+        self.serviceName = options.serviceName; // the application name for data lineage
     } else {
-        if ('applicationNameEnv' in options) {
-            self.applicationNameEnv = options.applicationNameEnv;
+        if ('serviceNameEnv' in options) {
+            self.serviceNameEnv = options.serviceNameEnv;
         } else {
-            self.applicationNameEnv = defaultAppNameEnvVar; // the Uber default app id
+            self.serviceNameEnv = defaultAppNameEnvVar; // the Uber default app id
         }
         /* eslint-disable no-process-env */
-        self.applicationName = process.env[self.applicationNameEnv];
+        self.serviceName = process.env[self.serviceNameEnv];
         /* eslint-enable no-process-env */
 
-        if (String.isNullOrEmpty(self.applicationName)) {
-            self.applicationName = utils.getApplicationName();
+        if (String.isNullOrEmpty(self.serviceName)) {
+            self.serviceName = utils.getServiceName();
         }
     }
 
-    if ('lineageHeader' in options) {
-        self.lineageHeader = options.lineageHeader;
+    if ('serviceNameHeader' in options) {
+        self.serviceNameHeader = options.serviceNameHeader;
     } else {
-        self.lineageHeader = defaultLineageHeaderName;
+        self.serviceNameHeader = defaultServiceNameHeader;
     }
 
     if ('clientVersionHeader' in options) {
         self.clientVersionHeader = options.clientVersionHeader;
     } else {
-        self.clientVersionHeader = defaultClientVersionHeader;
+        self.clientVersionHeader = defaultVersionHeader;
+    }
+
+    if ('instanceNameHeader' in options) {
```

Comment:
fyi, in the other libs, we allow user to provide an option called KafkaClientID which is passed to instanceName. But those libs infer serviceNames instead of allowing users to provide the value. The perception of service/instance pair is not exposed. Thus KafkaClientID may makes sense there.